### PR TITLE
SauceLabs setup

### PR DESCRIPTION
**Pull request by megawac**
TODOs in subsequent pull requests:

- [ ] run tests on IE6,7,8 (currently unsupported by Karma)
- [ ] clean up the travis file (develop a karma plugin to limit how many browsers they start at a single time, they have no interest in limiting based on IRC chats)

This is a continuation and resolves #1835 

**Do not merge until travis passes** (I'll work on this again tongiht)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,15 +7,25 @@ matrix:
   include:
     - node_js: "0.10"
       env: BROWSER=true
-env: BROWSER=false
 before_install:
   - npm install -g npm
-  - if [[ $BROWSER == true ]]; then npm install karma-phantomjs-launcher karma-firefox-launcher; fi
+  - npm install -g karma-cli
```
Comment (braddunbar) [inline .travis.yml]: Why would we install it globally instead of in `.node_modules`? The executable will still be in the PATH when running via `npm test`.

Comment (megawac) [inline .travis.yml]: I'm not sure that `karma` would be added to path like that seeing the commands aren't being run in `npm`s context. Either way, it doesn't really matter as its only being installed in travis sandbox

**Comment by megawac** [comment]
IT WORKS! https://travis-ci.org/jashkenas/underscore/jobs/51519224

**Comment by jridgewell** [comment]
:balloon: This is sorely needed, especially after yesterday. :+1:

**Comment by megawac** [comment]
Sweet its working after #2069. Everyone happy with this?

File: .travis.yml [hunk 2]
```diff
@@ -9,15 +7,26 @@ matrix:
...
+  - "[ $BROWSER == false ] || karma start karma.conf-sauce.js --browsers FIREFOX_V4,FIREFOX_V11,FIREFOX_V20,FIREFOX_V30,FIREFOX_V35"
+  - "[ $BROWSER == false ] || karma start karma.conf-sauce.js --browsers CHROME_V28,CHROME_V35,CHROME_V40,ANDROID_V4.0,ANDROID_V4.3"
+  - "[ $BROWSER == false ] || karma start karma.conf-sauce.js --browsers INTERNET_EXPLORER_V9,INTERNET_EXPLORER_V10,INTERNET_EXPLORER_V11"
+  - "[ $BROWSER == false ] || karma start karma.conf-sauce.js --browsers SAFARI_V5,SAFARI_V6,SAFARI_V7"
+  - "[ $BROWSER == false ] || karma start karma.conf-sauce.js --browsers OPERA_V11,OPERA_V12"
 notifications:
   email: false
+env:
+  global:
+  - secure: bDZSBQfqr21hCayjcZ20IxrV6+XGhxQPFIfwWqEKLrF93Gu8LLVjZRxXE/mE8I8N4Z5WtDNb4ZHrm/TTzmcPa5MuHgIxEdknQCncobH8oimwc83SHwEPk6okeNKl39VlCjvvnmoe/V/KpnknuYn3Rqghtl/Uv9KLpCwskwjTtcw=
```
Comment (michaelficarra) [inline .travis.yml]: What is this magic?

**Comment by jashkenas** [comment]
Bravo!

File: .travis.yml (hunk 2 above)
Comment (megawac) [inline .travis.yml]: Its the encrypted sauce token

Comment (michaelficarra) [inline .travis.yml]: What does that mean? What does it do? Why are there two?

Comment (megawac) [inline .travis.yml]: @michaelficarra see https://docs.saucelabs.com/ci-integrations/travis-ci/

Comment (michaelficarra) [inline .travis.yml]: So whose credentials are these, then? Can't anybody just re-use them on their own repo, or are they unique per repo? They don't seem to be unique, given how they are generated.

Comment (jridgewell) [inline .travis.yml]: They're encrypted with a repo specific Travis key, so only underscore can use these. @megawac are these your login credentials?

Comment (megawac) [inline .travis.yml]: @jashkenas created the account months ago, do you need the credentials?

Comment (megawac) [inline .travis.yml]: @michaelficarra and yeah travis uses the repo name `jashkenas/underscore` to generate those keys

Comment (jridgewell) [inline .travis.yml]: No, I was just curious if they were your own credentials.

**Pull request merged**